app/views/browse/pod.html

```diff
@@ -119,24 +119,68 @@
-                      <span class="pficon pficon-info" aria-hidden="true"></span>
-                      When you navigate away from this pod, any processes running
-                      in these terminals will quit.
-                    </div>
-                    <div class="pod-container-terminal"
-                      ng-repeat="container in containers | orderBy:'name' track by container.name"
-                      ng-if="container.state.running">
-                      <h3 ng-if="containers.length > 1">{{container.name}}</h3>
-                      <kubernetes-container-terminal pod="pod" container="container.name"
-                        prevent="!terminalTabWasSelected">
-                      </kubernetes-container-terminal>
-                    </div>
+                    ng-if="containersRunning(pod.status.containerStatuses) > 0">
+                      <uib-tab-heading>Terminal</uib-tab-heading>
+
+                      <div class="mar-bottom-md">
+                        <span class="pficon pficon-info" aria-hidden="true"></span>
+                        When you navigate away from this pod, any open terminal connections will be closed.
+                        This will kill any foreground processes you started from the terminal.
+                      </div>
+
+                      <div class="mar-bottom-lg">
+                        <div class="pad-left-none col-xs-4 col-lg-3">
+                          <ui-select
+                            ng-model="selectedTerminalContainer"
+                            on-select="onTerminalSelectChange(selectedTerminalContainer)"
+                            class="mar-left-none pad-left-none pad-right-none">
+                           <ui-select-match
+                              class="truncate"
+                              placeholder="Container Name">
+                                <span class="pad-left-md">
+                                  {{selectedTerminalContainer.containerName}}
+                                </span>
+                            </ui-select-match>
+                            <ui-select-choices
+                              repeat="term in containerTerminals | filter: $select.search"
+                              ui-disable-choice="term.containerStatus === 'waiting'">
+                              <div row>
+                                <span ng-bind-html="term.containerName | highlight: $select.search">
+                                </span>
+                                <span flex></span>
+                                <span
+                                  ng-if="term.isUsed && term.containerStatus === 'running'"
+                                  ng-class="{'text-muted' : (term.status === 'disconnected')}">
+                                    {{term.status}}
+                                </span>
+                                <span
+                                  ng-if="term.isUsed && term.containerStatus !== 'running'"
```

Comment:
Suggest showing waiting or terminated state here even if you haven't picked the terminal (but not running).